tests/test_notify_suppliers_of_awarded_briefs.py

```diff
@@ -0,0 +1,192 @@
+import mock
+
+from freezegun import freeze_time
+
+from dmscripts import notify_suppliers_of_awarded_briefs as tested_script
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+
+
+AWARDED_BRIEFS = [
+    {
+        "framework": {
+            "family": "digital-outcomes-and-specialists",
+            "slug": "digital-outcomes-and-specialists-2",
+        },
+        "id": 123,
+        "status": "awarded",
+        "title": "Tea Drinker"
+    },
+    {
+        "framework": {
+            "family": "digital-outcomes-and-specialists",
+            "slug": "digital-outcomes-and-specialists-2",
+        },
+        "id": 456,
+        "status": "awarded",
+        "title": "Cookie Muncher"
+    }
+]
+
+
+EXPECTED_BRIEF_CONTEXT = {
+    'brief_title': "Tea Drinker",
+    'brief_link': 'https://www.preview.marketplace.team/digital-outcomes-and-specialists/opportunities/123'
+}
+
+
+def _get_dummy_brief_response(id_, brief, awarded=False, valid_email=True):
+    brief_response = {
+        "awardedAt": '2018-01-01T23:59:59.999999Z',
+        "briefId": brief['id'],
+        "brief": brief,
+        "id": id_,
+        "respondToEmailAddress": "sore_loser_{}@example.com".format(id_),
+    }
+    if awarded:
+        brief_response['awardedAt'] = '2018-01-01T23:59:59.999999Z'
+        brief_response['status'] = 'awarded'
+        brief_response["respondToEmailAddress"] = "lucky_winner_{}@example.com".format(id_)
+    if not valid_email:
+        brief_response['respondToEmailAddress'] = ""
+    return brief_response
+
+
+def test_create_context_for_brief():
+    assert tested_script._create_context_for_brief('preview', AWARDED_BRIEFS[0]) == EXPECTED_BRIEF_CONTEXT
+
+
+@mock.patch('dmutils.email.DMNotifyClient', autospec=True)
+@mock.patch('dmscripts.notify_suppliers_of_awarded_briefs._create_context_for_brief')
+@mock.patch('dmapiclient.DataAPIClient', autospec=True)
+def test_main_calls_correct_methods(data_api_client, create_context_for_brief, notify_client):
+    data_api_client.find_brief_responses_iter.side_effect = [
+        [   # Only get awarded BRs on the first call
+            _get_dummy_brief_response(4321, AWARDED_BRIEFS[0], awarded=True),
+            _get_dummy_brief_response(4322, AWARDED_BRIEFS[1], awarded=True),
+        ],
+        [   # Get submitted BRs for 1st awarded brief on the second call
+            _get_dummy_brief_response(4322, AWARDED_BRIEFS[0]),
+            _get_dummy_brief_response(4323, AWARDED_BRIEFS[0], valid_email=False),
+        ],
+        [   # No submitted BRs for the 2nd awarded brief
+        ]
+    ]
+    tested_script._create_context_for_brief.return_value = EXPECTED_BRIEF_CONTEXT
+
+    with freeze_time('2018-01-02'):
+        assert tested_script.main(data_api_client, notify_client, "notify_template_id", "preview", mock.Mock())
```

Comment:
Interestingly (?) you're mocking the _class_ and then passing that in where the script would expect an _instance_ to be passed in. But it seems to work so I guess that's ok (?)

Reply:
Yeah, it's an interesting one. As we initialise the clients in the top level function we never assert the mocked classes directly, only their methods. We've followed the pattern used in our other recent scripts here by initialising clients at top level - I'm on the fence as to whether this is better off done in the testable code.